--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,7 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
 option(BUILD_CXXLIBS "Build libFLAC++" ON)
 option(BUILD_EXAMPLES "Build and install examples" ON)
+option(BUILD_APPS "Build applications" ON)
 option(WITH_OGG "ogg support (default: test for libogg)" ON)
 
 if(WITH_OGG)
@@ -95,7 +96,9 @@ endif()
 
 add_subdirectory("doc")
 add_subdirectory("src")
-add_subdirectory("microbench")
+if(BUILD_APPS)
+  add_subdirectory("microbench")
+endif()
 if(BUILD_EXAMPLES)
     add_subdirectory("examples")
 endif()
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 3.0)
 
 option(ENABLE_64_BIT_WORDS "Set FLAC__BYTES_PER_WORD to 8 (4 is the default)" OFF)
 option(WITH_XMMS "Build XMMS plugin" OFF)
+option(WITH_APPS "Build applications" ON)
 
 check_include_file("iconv.h" HAVE_ICONV)
 
@@ -9,10 +10,13 @@ add_subdirectory("libFLAC")
 if(BUILD_CXXLIBS)
     add_subdirectory("libFLAC++")
 endif()
-add_subdirectory("share")
-add_subdirectory("flac")
-add_subdirectory("metaflac")
-add_subdirectory("utils")
+
+if(BUILD_APPS)
+  add_subdirectory("share")
+  add_subdirectory("flac")
+  add_subdirectory("metaflac")
+  add_subdirectory("utils")
+endif()
 
 if(WITH_XMMS)
     add_subdirectory("plugin_common")
