
diff -upr oldtree/drivers/media/dvb-core/dvb_net.c /home/v4l/tokernel/wrk/linux-next/drivers/media/dvb-core/dvb_net.c
--- next/drivers/media/dvb-core/dvb_net.c	2010-01-15 19:05:35.000000000 -0200
+++ ../oldtree/drivers/media/dvb-core/dvb_net.c	2010-01-15 19:11:43.000000000 -0200
@@ -949,11 +949,8 @@ static int dvb_net_filter_sec_set(struct
 	(*secfilter)->filter_mask[10] = mac_mask[1];
 	(*secfilter)->filter_mask[11]=mac_mask[0];
 
+	dprintk("%s: filter mac=%pM\n", dev->name, mac);
+	dprintk("%s: filter mask=%pM\n", dev->name, mac_mask);
-	dprintk("%s: filter mac=%02x %02x %02x %02x %02x %02x\n",
-	       dev->name, mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
-	dprintk("%s: filter mask=%02x %02x %02x %02x %02x %02x\n",
-	       dev->name, mac_mask[0], mac_mask[1], mac_mask[2],
-	       mac_mask[3], mac_mask[4], mac_mask[5]);
 
 	return 0;
 }
