Whitespace cleanup

--- git/sound/pci/bt87x.c	2007-10-16 23:42:25.000000000 -0200
+++ ../oldtree/sound/pci/bt87x.c	2007-10-17 12:46:18.000000000 -0200
@@ -122,8 +122,8 @@ MODULE_PARM_DESC(load_all, "Allow to loa
 /* RISC instruction bits */
 #define RISC_BYTES_ENABLE	(0xf << 12)	/* byte enable bits */
 #define RISC_RESYNC		(  1 << 15)	/* disable FDSR errors */
-#define RISC_SET_STATUS_SHIFT	        16	/* set status bits */
-#define RISC_RESET_STATUS_SHIFT	        20	/* clear status bits */
+#define RISC_SET_STATUS_SHIFT		16	/* set status bits */
+#define RISC_RESET_STATUS_SHIFT		20	/* clear status bits */
 #define RISC_IRQ		(  1 << 24)	/* interrupt */
 #define RISC_EOL		(  1 << 26)	/* end of line */
 #define RISC_SOL		(  1 << 27)	/* start of line */
@@ -226,7 +226,7 @@ static inline void snd_bt87x_writel(stru
 }
 
 static int snd_bt87x_create_risc(struct snd_bt87x *chip, struct snd_pcm_substream *substream,
-			       	 unsigned int periods, unsigned int period_bytes)
+				 unsigned int periods, unsigned int period_bytes)
 {
 	struct snd_sg_buf *sgbuf = snd_pcm_substream_sgbuf(substream);
 	unsigned int i, offset;
