diff --git a/drivers/media/tuners/tda18271-common.c b/drivers/media/tuners/tda18271-common.c
index 86e5e31..9f7a0eb 100644
--- a/drivers/media/tuners/tda18271-common.c
+++ b/drivers/media/tuners/tda18271-common.c
@@ -713,26 +713,3 @@ int tda18271_calc_rf_cal(struct dvb_frontend *fe, u32 *freq)
 fail:
 	return ret;
 }
-
-void _tda_printk(struct tda18271_priv *state, const char *level,
-		 const char *func, const char *fmt, ...)
-{
-	struct va_format vaf;
-	va_list args;
-
-	va_start(args, fmt);
-
-	vaf.fmt = fmt;
-	vaf.va = &args;
-
-	if (state)
-		printk("%s%s: [%d-%04x|%c] %pV",
-		       level, func, i2c_adapter_id(state->i2c_props.adap),
-		       state->i2c_props.addr,
-		       (state->role == TDA18271_MASTER) ? 'M' : 'S',
-		       &vaf);
-	else
-		printk("%s%s: %pV", level, func, &vaf);
-
-	va_end(args);
-}
diff --git a/drivers/media/tuners/tda18271-priv.h b/drivers/media/tuners/tda18271-priv.h
index b36a7b7..3bc5b1c 100644
--- a/drivers/media/tuners/tda18271-priv.h
+++ b/drivers/media/tuners/tda18271-priv.h
@@ -138,12 +138,17 @@ extern int tda18271_debug;
 #define DBG_ADV  8
 #define DBG_CAL  16
 
-__attribute__((format(printf, 4, 5)))
-void _tda_printk(struct tda18271_priv *state, const char *level,
-		 const char *func, const char *fmt, ...);
-
-#define tda_printk(st, lvl, fmt, arg...)			\
-	_tda_printk(st, lvl, __func__, fmt, ##arg)
+#define tda_printk(st, kern, fmt, arg...) do {\
+	if (st) { \
+		struct tda18271_priv *state = st; \
+		printk(kern "%s: [%d-%04x|%s] " fmt, __func__, \
+		i2c_adapter_id(state->i2c_props.adap), \
+		state->i2c_props.addr, \
+		(state->role == TDA18271_MASTER) \
+			? "M" : "S", ##arg); \
+	} else \
+		printk(kern "%s: " fmt, __func__, ##arg); \
+} while (0)
 
 #define tda_dprintk(st, lvl, fmt, arg...)			\
 do {								\
