diff --git a/drivers/media/rc/rc-main.c b/drivers/media/rc/rc-main.c
index 2424946..bcede3b 100644
--- a/drivers/media/rc/rc-main.c
+++ b/drivers/media/rc/rc-main.c
@@ -1591,8 +1591,8 @@ struct rc_dev *rc_allocate_device(enum rc_driver_type type)
 			return NULL;
 		}
 
-		dev->input_dev->getkeycode = ir_getkeycode;
-		dev->input_dev->setkeycode = ir_setkeycode;
+		dev->input_dev->getkeycode_new = ir_getkeycode;
+		dev->input_dev->setkeycode_new = ir_setkeycode;
 		input_set_drvdata(dev->input_dev, dev);
 
 		setup_timer(&dev->timer_keyup, ir_timer_keyup,
diff --git a/drivers/media/usb/dvb-usb/dvb-usb-remote.c b/drivers/media/usb/dvb-usb/dvb-usb-remote.c
index 059ded5..6bf4dd0 100644
--- a/drivers/media/usb/dvb-usb/dvb-usb-remote.c
+++ b/drivers/media/usb/dvb-usb/dvb-usb-remote.c
@@ -198,8 +198,8 @@ static int legacy_dvb_usb_remote_init(struct dvb_usb_device *d)
 	d->input_dev = input_dev;
 	d->rc_dev = NULL;
 
-	input_dev->getkeycode = legacy_dvb_usb_getkeycode;
-	input_dev->setkeycode = legacy_dvb_usb_setkeycode;
+	input_dev->getkeycode_new = legacy_dvb_usb_getkeycode;
+	input_dev->setkeycode_new = legacy_dvb_usb_setkeycode;
 
 	/* set the bits for the keys */
 	deb_rc("key map size: %d\n", d->props.rc.legacy.rc_map_size);
