diff --git a/drivers/media/dvb-core/dvb_net.c b/drivers/media/dvb-core/dvb_net.c
index f91c80c..d487c15 100644
--- a/drivers/media/dvb-core/dvb_net.c
+++ b/drivers/media/dvb-core/dvb_net.c
@@ -653,7 +653,7 @@ static void dvb_net_ule( struct net_device *dev, const u8 *buf, size_t buf_len )
 				dev_kfree_skb(priv->ule_skb);
 			} else {
 				/* CRC32 verified OK. */
-				u8 dest_addr[ETH_ALEN];
+				u8 dest_addr[ETH_ALEN] = { 0 };
 				static const u8 bc_addr[ETH_ALEN] =
 					{ [ 0 ... ETH_ALEN-1] = 0xff };
 
