diff --git a/drivers/media/dvb-frontends/cxd2820r_core.c b/drivers/media/dvb-frontends/cxd2820r_core.c
index 95267c6..6583f29 100644
--- a/drivers/media/dvb-frontends/cxd2820r_core.c
+++ b/drivers/media/dvb-frontends/cxd2820r_core.c
@@ -726,8 +726,10 @@ static int cxd2820r_remove(struct i2c_client *client)
 	dev_dbg(&client->dev, "\n");
 
 #ifdef CONFIG_GPIOLIB
-	if (priv->gpio_chip.label)
-		gpiochip_remove(&priv->gpio_chip);
+	if (priv->gpio_chip.label) {
+		int ret = gpiochip_remove(&priv->gpio_chip);
+		(void)ret;
+	}
 #endif
 	regmap_exit(priv->regmap[1]);
 	i2c_unregister_device(priv->client[1]);
