diff --git a/drivers/media/rc/lirc_dev.c b/drivers/media/rc/lirc_dev.c
index f862f1b..79f35b7 100644
--- a/drivers/media/rc/lirc_dev.c
+++ b/drivers/media/rc/lirc_dev.c
@@ -848,27 +848,4 @@ void __exit lirc_dev_exit(void)
 	unregister_chrdev_region(lirc_base_dev, RC_DEV_MAX);
 }
 
-struct rc_dev *rc_dev_get_from_fd(int fd)
-{
-	struct fd f = fdget(fd);
-	struct lirc_fh *fh;
-	struct rc_dev *dev;
-
-	if (!f.file)
-		return ERR_PTR(-EBADF);
-
-	if (f.file->f_op != &lirc_fops) {
-		fdput(f);
-		return ERR_PTR(-EINVAL);
-	}
-
-	fh = f.file->private_data;
-	dev = fh->rc;
-
-	get_device(&dev->dev);
-	fdput(f);
-
-	return dev;
-}
-
 MODULE_ALIAS("lirc_dev");
diff --git a/drivers/media/rc/rc-core-priv.h b/drivers/media/rc/rc-core-priv.h
index e847bda..389b528 100644
--- a/drivers/media/rc/rc-core-priv.h
+++ b/drivers/media/rc/rc-core-priv.h
@@ -13,7 +13,6 @@
 #define	MAX_IR_EVENT_SIZE	512
 
 #include <linux/slab.h>
-#include <uapi/linux/bpf.h>
 #include <media/rc-core.h>
 
 /**
