diff --git a/drivers/media/dvb-frontends/drx39xyj/drxj.c b/drivers/media/dvb-frontends/drx39xyj/drxj.c
index 7ca7a21..0ffd69b 100644
--- a/drivers/media/dvb-frontends/drx39xyj/drxj.c
+++ b/drivers/media/dvb-frontends/drx39xyj/drxj.c
@@ -10666,8 +10666,8 @@ ctrl_sig_quality(struct drx_demod_instance *demod,
 	struct dtv_frontend_properties *p = &state->frontend.dtv_property_cache;
 	enum drx_standard standard = ext_attr->standard;
 	int rc;
-	u32 ber, cnt, err, pkt;
-	u16 mer, strength = 0;
+	u32 ber = 0, cnt, err = 0, pkt;
+	u16 mer = 0, strength = 0;
 
 	rc = get_sig_strength(demod, &strength);
 	if (rc < 0) {
