diff --git a/drivers/media/pci/ivtv/ivtvfb.c b/drivers/media/pci/ivtv/ivtvfb.c
index cfd21040d0e3..a0d61a9e4459 100644
--- a/drivers/media/pci/ivtv/ivtvfb.c
+++ b/drivers/media/pci/ivtv/ivtvfb.c
@@ -1171,7 +1171,7 @@ static int ivtvfb_init_card(struct ivtv *itv)
 	int rc;
 
 #ifdef CONFIG_X86_64
-	if (pat_enabled()) {
+	if (pat_enabled) {
 		if (ivtvfb_force_pat) {
 			pr_info("PAT is enabled. Write-combined framebuffer caching will be disabled.\n");
 			pr_info("To enable caching, boot with nopat kernel parameter\n");
