diff --git a/drivers/media/tuners/tuner-xc2028.c b/drivers/media/tuners/tuner-xc2028.c
index f88f948..8591799 100644
--- a/drivers/media/tuners/tuner-xc2028.c
+++ b/drivers/media/tuners/tuner-xc2028.c
@@ -1408,7 +1408,6 @@ static int xc2028_set_config(struct dvb_frontend *fe, void *priv_cfg)
 		rc = request_firmware_nowait(THIS_MODULE, 1,
 					     priv->fname,
 					     priv->i2c_props.adap->dev.parent,
-					     GFP_KERNEL,
 					     fe, load_firmware_cb);
 		if (rc < 0) {
 			tuner_err("Failed to request firmware %s\n",
