---
 drivers/media/usb/pvrusb2/pvrusb2-sysfs.c |   14 --------------
 1 file changed, 14 deletions(-)

--- linux.orig/drivers/media/usb/pvrusb2/pvrusb2-sysfs.c
+++ linux/drivers/media/usb/pvrusb2/pvrusb2-sysfs.c
@@ -432,12 +432,10 @@ static void pvr2_sysfs_add_debugifc(stru
 
 	dip = kzalloc(sizeof(*dip),GFP_KERNEL);
 	if (!dip) return;
-	sysfs_attr_init(&dip->attr_debugcmd.attr);
 	dip->attr_debugcmd.attr.name = "debugcmd";
 	dip->attr_debugcmd.attr.mode = S_IRUGO|S_IWUSR|S_IWGRP;
 	dip->attr_debugcmd.show = debugcmd_show;
 	dip->attr_debugcmd.store = debugcmd_store;
-	sysfs_attr_init(&dip->attr_debuginfo.attr);
 	dip->attr_debuginfo.attr.name = "debuginfo";
 	dip->attr_debuginfo.attr.mode = S_IRUGO;
 	dip->attr_debuginfo.show = debuginfo_show;
@@ -659,8 +657,6 @@ static void class_dev_create(struct pvr2
 		put_device(class_dev);
 		return;
 	}
-
-	sysfs_attr_init(&sfp->attr_v4l_minor_number.attr);
 	sfp->attr_v4l_minor_number.attr.name = "v4l_minor_number";
 	sfp->attr_v4l_minor_number.attr.mode = S_IRUGO;
 	sfp->attr_v4l_minor_number.show = v4l_minor_number_show;
@@ -674,8 +670,6 @@ static void class_dev_create(struct pvr2
 	} else {
 		sfp->v4l_minor_number_created_ok = !0;
 	}
-
-	sysfs_attr_init(&sfp->attr_v4l_radio_minor_number.attr);
 	sfp->attr_v4l_radio_minor_number.attr.name = "v4l_radio_minor_number";
 	sfp->attr_v4l_radio_minor_number.attr.mode = S_IRUGO;
 	sfp->attr_v4l_radio_minor_number.show = v4l_radio_minor_number_show;
@@ -689,8 +683,6 @@ static void class_dev_create(struct pvr2
 	} else {
 		sfp->v4l_radio_minor_number_created_ok = !0;
 	}
-
-	sysfs_attr_init(&sfp->attr_unit_number.attr);
 	sfp->attr_unit_number.attr.name = "unit_number";
 	sfp->attr_unit_number.attr.mode = S_IRUGO;
 	sfp->attr_unit_number.show = unit_number_show;
@@ -703,8 +695,6 @@ static void class_dev_create(struct pvr2
 	} else {
 		sfp->unit_number_created_ok = !0;
 	}
-
-	sysfs_attr_init(&sfp->attr_bus_info.attr);
 	sfp->attr_bus_info.attr.name = "bus_info_str";
 	sfp->attr_bus_info.attr.mode = S_IRUGO;
 	sfp->attr_bus_info.show = bus_info_show;
@@ -718,8 +708,6 @@ static void class_dev_create(struct pvr2
 	} else {
 		sfp->bus_info_created_ok = !0;
 	}
-
-	sysfs_attr_init(&sfp->attr_hdw_name.attr);
 	sfp->attr_hdw_name.attr.name = "device_hardware_type";
 	sfp->attr_hdw_name.attr.mode = S_IRUGO;
 	sfp->attr_hdw_name.show = hdw_name_show;
@@ -733,8 +721,6 @@ static void class_dev_create(struct pvr2
 	} else {
 		sfp->hdw_name_created_ok = !0;
 	}
-
-	sysfs_attr_init(&sfp->attr_hdw_desc.attr);
 	sfp->attr_hdw_desc.attr.name = "device_hardware_description";
 	sfp->attr_hdw_desc.attr.mode = S_IRUGO;
 	sfp->attr_hdw_desc.show = hdw_desc_show;
