diff --git a/drivers/media/dvb-core/dvb_net.c b/drivers/media/dvb-core/dvb_net.c
index 686d327..eabf882 100644
--- a/drivers/media/dvb-core/dvb_net.c
+++ b/drivers/media/dvb-core/dvb_net.c
@@ -1087,14 +1087,14 @@ static int dvb_net_feed_stop(struct net_device *dev)
 }
 
 
-static int dvb_set_mc_filter(struct net_device *dev, unsigned char *addr)
+static int dvb_set_mc_filter (struct net_device *dev, struct dev_mc_list *mc)
 {
 	struct dvb_net_priv *priv = netdev_priv(dev);
 
 	if (priv->multi_num == DVB_NET_MULTICAST_MAX)
 		return -ENOMEM;
 
-	memcpy(priv->multi_macs[priv->multi_num], addr, ETH_ALEN);
+	memcpy(priv->multi_macs[priv->multi_num], mc->dmi_addr, 6);
 
 	priv->multi_num++;
 	return 0;
@@ -1118,7 +1118,7 @@ static void wq_set_multicast_list (struct work_struct *work)
 		netdev_dbg(dev, "allmulti mode\n");
 		priv->rx_mode = RX_MODE_ALL_MULTI;
 	} else if (!netdev_mc_empty(dev)) {
-		struct netdev_hw_addr *ha;
+		struct dev_mc_list *mc;
 
 		netdev_dbg(dev, "set_mc_list, %d entries\n",
 			   netdev_mc_count(dev));
@@ -1126,8 +1126,8 @@ static void wq_set_multicast_list (struct work_struct *work)
 		priv->rx_mode = RX_MODE_MULTI;
 		priv->multi_num = 0;
 
-		netdev_for_each_mc_addr(ha, dev)
-			dvb_set_mc_filter(dev, ha->addr);
+		netdev_for_each_mc_addr(mc, dev)
+			dvb_set_mc_filter(dev, mc);
 	}
 
 	netif_addr_unlock_bh(dev);
