---
 drivers/media/firewire/firedtv-fw.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- linux.orig/drivers/media/firewire/firedtv-fw.c
+++ linux/drivers/media/firewire/firedtv-fw.c
@@ -192,9 +192,9 @@ void fdtv_stop_iso(struct firedtv *fdtv)
 }
 
 static void handle_fcp(struct fw_card *card, struct fw_request *request,
-		       int tcode, int destination, int source, int generation,
-		       unsigned long long offset, void *payload, size_t length,
-		       void *callback_data)
+                       int tcode, int destination, int source, int generation,
+                       int speed, unsigned long long offset,
+                       void *payload, size_t length, void *callback_data)
 {
 	struct firedtv *f, *fdtv = NULL;
 	struct fw_device *device;
