diff --git a/drivers/media/i2c/as3645a.c b/drivers/media/i2c/as3645a.c
index 301084b..1776383 100644
--- a/drivers/media/i2c/as3645a.c
+++ b/drivers/media/i2c/as3645a.c
@@ -863,7 +863,7 @@ static const struct i2c_device_id as3645a_id_table[] = {
 };
 MODULE_DEVICE_TABLE(i2c, as3645a_id_table);
 
-static const struct dev_pm_ops as3645a_pm_ops = {
+static struct dev_pm_ops as3645a_pm_ops = {
 	.suspend = as3645a_suspend,
 	.resume = as3645a_resume,
 };
diff --git a/drivers/media/i2c/msp3400-driver.c b/drivers/media/i2c/msp3400-driver.c
index 8190fec..9ed1198c 100644
--- a/drivers/media/i2c/msp3400-driver.c
+++ b/drivers/media/i2c/msp3400-driver.c
@@ -882,7 +882,7 @@ static int msp_remove(struct i2c_client *client)
 
 /* ----------------------------------------------------------------------- */
 
-static const struct dev_pm_ops msp3400_pm_ops = {
+static struct dev_pm_ops msp3400_pm_ops = {
 	SET_SYSTEM_SLEEP_PM_OPS(msp_suspend, msp_resume)
 };
 
diff --git a/drivers/media/pci/meye/meye.c b/drivers/media/pci/meye/meye.c
index 2381b05..09b4a49 100644
--- a/drivers/media/pci/meye/meye.c
+++ b/drivers/media/pci/meye/meye.c
@@ -1461,7 +1461,7 @@ static void meye_vm_close(struct vm_area_struct *vma)
 	meye.vma_use_count[idx]--;
 }
 
-static const struct vm_operations_struct meye_vm_ops = {
+static struct vm_operations_struct meye_vm_ops = {
 	.open		= meye_vm_open,
 	.close		= meye_vm_close,
 };
diff --git a/drivers/media/usb/gspca/gspca.c b/drivers/media/usb/gspca/gspca.c
index b7ae872..30a450e 100644
--- a/drivers/media/usb/gspca/gspca.c
+++ b/drivers/media/usb/gspca/gspca.c
@@ -107,7 +107,7 @@ static void gspca_vm_close(struct vm_area_struct *vma)
 		frame->v4l2_buf.flags &= ~V4L2_BUF_FLAG_MAPPED;
 }
 
-static const struct vm_operations_struct gspca_vm_ops = {
+static struct vm_operations_struct gspca_vm_ops = {
 	.open		= gspca_vm_open,
 	.close		= gspca_vm_close,
 };
diff --git a/drivers/media/usb/stkwebcam/stk-webcam.c b/drivers/media/usb/stkwebcam/stk-webcam.c
index c43c8d3..6d14f20 100644
--- a/drivers/media/usb/stkwebcam/stk-webcam.c
+++ b/drivers/media/usb/stkwebcam/stk-webcam.c
@@ -747,7 +747,7 @@ static void stk_v4l_vm_close(struct vm_area_struct *vma)
 	if (sbuf->mapcount == 0)
 		sbuf->v4lbuf.flags &= ~V4L2_BUF_FLAG_MAPPED;
 }
-static const struct vm_operations_struct stk_v4l_vm_ops = {
+static struct vm_operations_struct stk_v4l_vm_ops = {
 	.open = stk_v4l_vm_open,
 	.close = stk_v4l_vm_close
 };
diff --git a/drivers/media/v4l2-core/tuner-core.c b/drivers/media/v4l2-core/tuner-core.c
index ddc9379..f4ef7f5 100644
--- a/drivers/media/v4l2-core/tuner-core.c
+++ b/drivers/media/v4l2-core/tuner-core.c
@@ -1324,7 +1324,7 @@ static const struct v4l2_subdev_ops tuner_ops = {
  * I2C structs and module init functions
  */
 
-static const struct dev_pm_ops tuner_pm_ops = {
+static struct dev_pm_ops tuner_pm_ops = {
 	SET_SYSTEM_SLEEP_PM_OPS(tuner_suspend, tuner_resume)
 };
 
diff --git a/drivers/media/v4l2-core/videobuf-dma-contig.c b/drivers/media/v4l2-core/videobuf-dma-contig.c
index 65411ad..7365adc 100644
--- a/drivers/media/v4l2-core/videobuf-dma-contig.c
+++ b/drivers/media/v4l2-core/videobuf-dma-contig.c
@@ -132,7 +132,7 @@ static void videobuf_vm_close(struct vm_area_struct *vma)
 	videobuf_queue_unlock(q);
 }
 
-static const struct vm_operations_struct videobuf_vm_ops = {
+static struct vm_operations_struct videobuf_vm_ops = {
 	.open	= videobuf_vm_open,
 	.close	= videobuf_vm_close,
 };
diff --git a/drivers/media/v4l2-core/videobuf-dma-sg.c b/drivers/media/v4l2-core/videobuf-dma-sg.c
index 9db674c..7cc4d4b 100644
--- a/drivers/media/v4l2-core/videobuf-dma-sg.c
+++ b/drivers/media/v4l2-core/videobuf-dma-sg.c
@@ -405,7 +405,7 @@ static int videobuf_vm_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
 	return 0;
 }
 
-static const struct vm_operations_struct videobuf_vm_ops = {
+static struct vm_operations_struct videobuf_vm_ops = {
 	.open	= videobuf_vm_open,
 	.close	= videobuf_vm_close,
 	.fault	= videobuf_vm_fault,
diff --git a/drivers/media/v4l2-core/videobuf-vmalloc.c b/drivers/media/v4l2-core/videobuf-vmalloc.c
index 1365c65..dfc24343 100644
--- a/drivers/media/v4l2-core/videobuf-vmalloc.c
+++ b/drivers/media/v4l2-core/videobuf-vmalloc.c
@@ -122,7 +122,7 @@ static void videobuf_vm_close(struct vm_area_struct *vma)
 	return;
 }
 
-static const struct vm_operations_struct videobuf_vm_ops = {
+static struct vm_operations_struct videobuf_vm_ops = {
 	.open     = videobuf_vm_open,
 	.close    = videobuf_vm_close,
 };
diff --git a/drivers/media/v4l2-core/videobuf2-memops.c b/drivers/media/v4l2-core/videobuf2-memops.c
index 81c1ad8..4912aa1 100644
--- a/drivers/media/v4l2-core/videobuf2-memops.c
+++ b/drivers/media/v4l2-core/videobuf2-memops.c
@@ -176,7 +176,7 @@ static void vb2_common_vm_close(struct vm_area_struct *vma)
  * vb2_common_vm_ops - common vm_ops used for tracking refcount of mmaped
  * video buffers
  */
-const struct vm_operations_struct vb2_common_vm_ops = {
+struct vm_operations_struct vb2_common_vm_ops = {
 	.open = vb2_common_vm_open,
 	.close = vb2_common_vm_close,
 };
diff --git a/include/media/videobuf2-memops.h b/include/media/videobuf2-memops.h
index f05444c..c395259 100644
--- a/include/media/videobuf2-memops.h
+++ b/include/media/videobuf2-memops.h
@@ -28,7 +28,7 @@ struct vb2_vmarea_handler {
 	void			*arg;
 };
 
-extern const struct vm_operations_struct vb2_common_vm_ops;
+extern struct vm_operations_struct vb2_common_vm_ops;
 
 int vb2_get_contig_userptr(unsigned long vaddr, unsigned long size,
 			   struct vm_area_struct **res_vma, dma_addr_t *res_pa);
