diff --git a/drivers/media/v4l2-core/v4l2-device.c b/drivers/media/v4l2-core/v4l2-device.c
index 9ed59da..2efb88a 100644
--- a/drivers/media/v4l2-core/v4l2-device.c
+++ b/drivers/media/v4l2-core/v4l2-device.c
@@ -130,8 +130,11 @@ void v4l2_device_unregister(struct v4l2_device *v4l2_dev)
 			 * registered by us, and would not be
 			 * re-created by just probing the V4L2 driver.
 			 */
-			if (client &&
-			    !client->dev.of_node)
+#ifdef CONFIG_OF
+			if (client && !client->dev.of_node)
+#else
+			if (client)
+#endif
 				i2c_unregister_device(client);
 			continue;
 		}
@@ -140,7 +143,11 @@ void v4l2_device_unregister(struct v4l2_device *v4l2_dev)
 		if (sd->flags & V4L2_SUBDEV_FL_IS_SPI) {
 			struct spi_device *spi = v4l2_get_subdevdata(sd);
 
+#ifdef CONFIG_OF
 			if (spi && !spi->dev.of_node)
+#else
+			if (spi)
+#endif
 				spi_unregister_device(spi);
 			continue;
 		}
