diff --git a/drivers/media/pci/saa7164/saa7164-encoder.c b/drivers/media/pci/saa7164/saa7164-encoder.c
index 1b184c3..3537c6f 100644
--- a/drivers/media/pci/saa7164/saa7164-encoder.c
+++ b/drivers/media/pci/saa7164/saa7164-encoder.c
@@ -389,7 +389,7 @@ int saa7164_s_frequency(struct saa7164_port *port,
 	if (f->tuner != 0)
 		return -EINVAL;
 
-	port->freq = clamp(f->frequency,
+	port->freq = clamp_t(u32, f->frequency,
 			   SAA7164_TV_MIN_FREQ, SAA7164_TV_MAX_FREQ);
 
 	/* Update the hardware */
diff --git a/drivers/media/usb/hackrf/hackrf.c b/drivers/media/usb/hackrf/hackrf.c
index e05bfec..b132ecb 100644
--- a/drivers/media/usb/hackrf/hackrf.c
+++ b/drivers/media/usb/hackrf/hackrf.c
@@ -1108,7 +1108,7 @@ static int hackrf_s_frequency(struct file *file, void *priv,
 			f->tuner, f->type, f->frequency);
 
 	if (f->tuner == 0) {
-		uitmp = clamp(f->frequency, bands_adc_dac[0].rangelow,
+		uitmp = clamp_t(u32, f->frequency, bands_adc_dac[0].rangelow,
 			      bands_adc_dac[0].rangehigh);
 		if (vdev->vfl_dir == VFL_DIR_RX) {
 			dev->f_adc = uitmp;
@@ -1118,7 +1118,7 @@ static int hackrf_s_frequency(struct file *file, void *priv,
 			set_bit(TX_DAC_FREQUENCY, &dev->flags);
 		}
 	} else if (f->tuner == 1) {
-		uitmp = clamp(f->frequency, bands_rx_tx[0].rangelow,
+		uitmp = clamp_t(u32, f->frequency, bands_rx_tx[0].rangelow,
 			      bands_rx_tx[0].rangehigh);
 		if (vdev->vfl_dir == VFL_DIR_RX) {
 			dev->f_rx = uitmp;
