--- a/drivers/media/firewire/firedtv-fw.c	2013-07-24 14:34:38.319443243 +0200
+++ b/drivers/media/firewire/firedtv-fw.c	2013-05-11 16:19:28.000000000 +0200
@@ -248,7 +248,7 @@
 /* Adjust the template string if models with longer names appear. */
 #define MAX_MODEL_NAME_LEN sizeof("FireDTV ????")
 
-static int node_probe(struct fw_unit *unit, const struct ieee1394_device_id *id)
+static int node_probe(struct device *dev)
 {
 	struct firedtv *fdtv;
 	char name[MAX_MODEL_NAME_LEN];
@@ -258,8 +258,8 @@
 	if (!fdtv)
 		return -ENOMEM;
 
-	dev_set_drvdata(&unit->device, fdtv);
-	fdtv->device		= &unit->device;
+	dev_set_drvdata(dev, fdtv);
+	fdtv->device		= dev;
 	fdtv->isochannel	= -1;
 	fdtv->voltage		= 0xff;
 	fdtv->tone		= 0xff;
@@ -269,7 +269,7 @@
 	mutex_init(&fdtv->demux_mutex);
 	INIT_WORK(&fdtv->remote_ctrl_work, avc_remote_ctrl_work);
 
-	name_len = fw_csr_string(unit->directory, CSR_MODEL,
+	name_len = fw_csr_string(fw_unit(dev)->directory, CSR_MODEL,
 				 name, sizeof(name));
 	for (i = ARRAY_SIZE(model_names); --i; )
 		if (strlen(model_names[i]) <= name_len &&
@@ -277,7 +277,7 @@
 			break;
 	fdtv->type = i;
 
-	err = fdtv_register_rc(fdtv, &unit->device);
+	err = fdtv_register_rc(fdtv, dev);
 	if (err)
 		goto fail_free;
 
@@ -307,9 +307,9 @@
 	return err;
 }
 
-static void node_remove(struct fw_unit *unit)
+static int node_remove(struct device *dev)
 {
-	struct firedtv *fdtv = dev_get_drvdata(&unit->device);
+	struct firedtv *fdtv = dev_get_drvdata(dev);
 
 	fdtv_dvb_unregister(fdtv);
 
@@ -320,6 +320,7 @@
 	fdtv_unregister_rc(fdtv);
 
 	kfree(fdtv);
+	return 0;
 }
 
 static void node_update(struct fw_unit *unit)
@@ -390,10 +391,10 @@
 		.owner  = THIS_MODULE,
 		.name   = "firedtv",
 		.bus    = &fw_bus_type,
+		.probe  = node_probe,
+		.remove = node_remove,
 	},
-	.probe    = node_probe,
 	.update   = node_update,
-	.remove   = node_remove,
 	.id_table = fdtv_id_table,
 };
 
