diff --git a/mm/frame_vector.c b/mm/frame_vector.c
index 7cf2b71..1199a59 100644
--- a/mm/frame_vector.c
+++ b/mm/frame_vector.c
@@ -58,8 +58,8 @@ int get_vaddr_frames(unsigned long start, unsigned int nr_frames,
 	if (!(vma->vm_flags & (VM_IO | VM_PFNMAP))) {
 		vec->got_ref = true;
 		vec->is_pfns = false;
-		ret = get_user_pages_locked(current, mm, start, nr_frames,
-			write, force, (struct page **)(vec->ptrs), &locked);
+		ret = get_user_pages(current, mm, start, nr_frames,
+			write, force, (struct page **)(vec->ptrs), NULL);
 		goto out;
 	}
 
