diff --git a/drivers/media/i2c/tda1997x.c b/drivers/media/i2c/tda1997x.c
index 039a92c..f244647 100644
--- a/drivers/media/i2c/tda1997x.c
+++ b/drivers/media/i2c/tda1997x.c
@@ -2492,7 +2492,7 @@ static struct snd_soc_component_driver tda1997x_codec_driver = {
 	.probe			= tda1997x_codec_probe,
 	.remove			= tda1997x_codec_remove,
 	.idle_bias_on		= 1,
-	.use_pmdown_time	= 1,
+	.pmdown_time		= 1,
 	.endianness		= 1,
 	.non_legacy_dai_naming	= 1,
 };
