diff --git a/drivers/media/pci/saa7164/saa7164-core.c b/drivers/media/pci/saa7164/saa7164-core.c
index f5ad3cf207d3..6962758c6765 100644
--- a/drivers/media/pci/saa7164/saa7164-core.c
+++ b/drivers/media/pci/saa7164/saa7164-core.c
@@ -1124,9 +1124,21 @@ static int saa7164_proc_show(struct seq_file *m, void *v)
 
 static struct proc_dir_entry *saa7164_pe;
 
+static int saa7164_proc_open(struct inode *inode, struct file *filp)
+{
+	return single_open(filp, saa7164_proc_show, NULL);
+}
+
+static const struct file_operations saa7164_proc_fops = {
+	.open		= saa7164_proc_open,
+	.read		= seq_read,
+	.llseek		= seq_lseek,
+	.release	= single_release,
+};
+
 static int saa7164_proc_create(void)
 {
-	saa7164_pe = proc_create_single("saa7164", 0444, NULL, saa7164_proc_show);
+	saa7164_pe = proc_create("saa7164", 0444, NULL, &saa7164_proc_fops);
 	if (!saa7164_pe)
 		return -ENOMEM;
 
