diff --git a/drivers/media/dvb-frontends/cxd2820r_core.c b/drivers/media/dvb-frontends/cxd2820r_core.c
index 95267c6..d980d6b 100644
--- a/drivers/media/dvb-frontends/cxd2820r_core.c
+++ b/drivers/media/dvb-frontends/cxd2820r_core.c
@@ -434,7 +434,8 @@ static int cxd2820r_i2c_gate_ctrl(struct dvb_frontend *fe, int enable)
 static int cxd2820r_gpio_direction_output(struct gpio_chip *chip, unsigned nr,
 		int val)
 {
-	struct cxd2820r_priv *priv = gpiochip_get_data(chip);
+	struct cxd2820r_priv *priv =
+			container_of(chip, struct cxd2820r_priv, gpio_chip);
 	struct i2c_client *client = priv->client[0];
 	u8 gpio[GPIO_COUNT];
 
@@ -448,7 +449,8 @@ static int cxd2820r_gpio_direction_output(struct gpio_chip *chip, unsigned nr,
 
 static void cxd2820r_gpio_set(struct gpio_chip *chip, unsigned nr, int val)
 {
-	struct cxd2820r_priv *priv = gpiochip_get_data(chip);
+	struct cxd2820r_priv *priv =
+			container_of(chip, struct cxd2820r_priv, gpio_chip);
 	struct i2c_client *client = priv->client[0];
 	u8 gpio[GPIO_COUNT];
 
@@ -464,7 +466,8 @@ static void cxd2820r_gpio_set(struct gpio_chip *chip, unsigned nr, int val)
 
 static int cxd2820r_gpio_get(struct gpio_chip *chip, unsigned nr)
 {
-	struct cxd2820r_priv *priv = gpiochip_get_data(chip);
+	struct cxd2820r_priv *priv =
+			container_of(chip, struct cxd2820r_priv, gpio_chip);
 	struct i2c_client *client = priv->client[0];
 
 	dev_dbg(&client->dev, "nr=%u\n", nr);
@@ -670,7 +673,7 @@ static int cxd2820r_probe(struct i2c_client *client,
 		priv->gpio_chip.base = -1; /* Dynamic allocation */
 		priv->gpio_chip.ngpio = GPIO_COUNT;
 		priv->gpio_chip.can_sleep = 1;
-		ret = gpiochip_add_data(&priv->gpio_chip, priv);
+		ret = gpiochip_add(&priv->gpio_chip);
 		if (ret)
 			goto err_regmap_1_regmap_exit;
 
